giraffe/src/utils/nearestTimestamp.ts

```diff
@@ -0,0 +1,25 @@
+export const nearestTimestamp = (timestamps, rawValue) => {
+  if (timestamps.length === 0) {
+    return rawValue
+  }
+
+  if (timestamps.length === 1) {
+    return timestamps[0]
+  }
+
+  const midPoint = timestamps.length / 2
+  const firstHalf = timestamps.slice(0, midPoint)
+  const secondHalf = timestamps.slice(midPoint)
```

Comment:
trading some memory and the cost of creating an array for code clarity. We only need to create one array of length `timestamps.length / 2` but this code creates two because it's much easier to follow